test/test_geometry.jl

```diff
@@ -243,17 +265,35 @@ import GeoFormatTypes as GFT
     end
 
     @testset "Testing construction of complex geometries" begin
-        @test AG.toWKT(
-                  AG.createlinestring(
-                      [1.0f0, 2.0f0, 3.0f0],
-                      [4.0f0, 5.0f0, 6.0f0],
-                  ),
-              ) ==
-              AG.toWKT(AG.createlinestring([1.0, 2.0, 3.0], [4.0, 5.0, 6.0])) ==
-              "LINESTRING (1 4,2 5,3 6)"
-        AG.createlinestring([1.0, 2.0, 3.0], [4.0, 5.0, 6.0]) do geom
+        @testset "linestring" begin
+            @test AG.toWKT(AG.createlinestring([1.0f0, 2.0f0, 3.0f0], [4.0f0, 5.0f0, 6.0f0])) ==
+                  AG.toWKT(AG.createlinestring([1.0, 2.0, 3.0], [4.0, 5.0, 6.0])) ==
+                  "LINESTRING (1 4,2 5,3 6)"
+            # create
+            AG.createlinestring([1.0, 2.0, 3.0], [4.0, 5.0, 6.0]) do geom
+                @test GI.geomtrait(geom) == GI.LineStringTrait()
+                @test GI.testgeometry(geom)
+                @test !AG.is3d(geom)
+                @test typeof(geom) == AG.Geometry{AG.wkbLineString}
+                @test typeof(AG.getgeom(geom, 0)) == AG.IGeometry{AG.wkbPoint}
+                @test isapprox(
+                    GI.coordinates(geom),
+                    [[1, 4], [2, 5], [3, 6]],
+                    atol = 1e-6,
+                )
+                @test AG.toWKT(geom) == "LINESTRING (1 4,2 5,3 6)"
+                AG.closerings!(geom)
+                @test AG.toWKT(geom) == "LINESTRING (1 4,2 5,3 6)"
+                AG.setpoint!(geom, 1, 10, 10)
+                @test AG.toWKT(geom) == "LINESTRING (1 4,10 10,3 6)"
+                @test GFT.val(convert(GFT.WellKnownText, geom)) == AG.toWKT(geom)
+            end
+            # unsafe_create
+            geom = AG.unsafe_createlinestring([1.0, 2.0, 3.0], [4.0, 5.0, 6.0])
```

Comment:
To my understanding, `AG.createlinestring(...)` is going to call `AG.unsafe_createlinestring(...)` [under-the-hood](https://github.com/yeesian/ArchGDAL.jl/blob/fe68c14e10a08a68be70c551a42c260e24c4da48/src/context.jl#L265), so I'm curious why we're duplicating the tests here.

If there is reason for doing so, can we indent all the code between `geom = AG.unsafe_createlinestring(...)` and `AG.destroy(geom)` so that it's clear where the scope of `geom` should be?

(Same comment(s) for `unsafe_createlinearring` below.)

Reply:
For linestrring hey're actually the same code that swaps in `create` and `unsafe_create` in an `@eval` loop:
https://github.com/rafaqz/ArchGDAL.jl/blob/stability/src/ogr/geometry.jl#L1714-L1720

IMO fine to test both of those.

Linearring actually has completely separate functions, because there is no GDAL enum for `wkbLinearRing25D` So we cant pass the type through, and have to rewrap with the correct `IGeometry{wkbLinearRing25D}`/`Geometry{wkbLinearRing}` 
afterwards. 

Probably they we could get rid of all of that complexity and have `create_x` always just call `unsafe_create_x` and add a finaliser to the result. I would like to get rid of a lot of that code generation, its pretty confusing and means you cant search or grep for method definitions in the code. But maybe thats  another PR.

Reply:
> Probably we could get rid of all of that complexity and have create_x always just call unsafe_create_x and add a finaliser to the result. I would like to get rid of a lot of that code generation, its pretty confusing and means you cant search or grep for method definitions in the code. But maybe thats another PR.

Yeah okay, sounds good; I understand it's all really knotty right now, thank you so much!